### PR TITLE
getting there/Saturday submission

### PR DESCRIPTION
**Pull request by spenc-neit**
Ran into a roadblock with the navbar - tried looking for tutorials, couldn't figure out how to apply them to my project.
The view all products page mostly works; everything displays, the links have the proper address, clicking them even changes the URL, but the page itself doesn't change. With some testing stuff I was able to make the product details page, so that as long as I can find a way to navigate there and give it the value it needs, it'll work perfectly.
I can't do the navigation properly yet though, and I also don't know how to approach adding the navbar to the page. I suspect all this has to with my handling of react router, which I don't have a good handle on yet.

Not sure how much you're "allowed" to help us out on this, but I figured I'd just send it in, ask for some pointers/help/tips, and call it a day (been working on this for like 9 hours now).

### CODE REVIEW

File: src/readAll.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { Text } from "./components/Text";
+import {
+	Link,
+	Route,
+	Routes,
+	useNavigate,
+	BrowserRouter,
+} from "react-router-dom";
+
+function ReadAll() {
```
Comment (bowen9284) [inline src/readAll.jsx:13]: Naming is important for readability so others understand what you are doing. I would rename this to `Products`, move it into the `components` directory and then delete `ProdView`. After the update your route would look something like this
```
<Route path="products" element={<Products />}>
```

File: src/main.jsx [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>,
-)
+ReactDOM.createRoot(document.getElementById("root")).render(
+	<React.StrictMode>
+		<BrowserRouter>
+			<Routes>
+				<Route index element={<App />} />
+				<Route path="prodview" element={<ReadAll />}>
```
Comment (bowen9284) [inline src/main.jsx:14]: Let's update this path to `path="products"`. There really isn't a concept of a `view` in React, almost everything is a component that contains JSX so they would all be views. The path in our route should describe what the path contains and exclude any technical terms whenever possible

File: src/OneEntry.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { useParams } from "react-router-dom";
+import { useFetch } from "./hooks/useFetch";
+import "./OneEntry.css"
+import { Box } from "./components/Box";
+import { Text } from "./components/Text";
+
+const OneEntry = () =>{
```
Comment (bowen9284) [inline src/OneEntry.jsx:7]: Suggestion for name change: `ProductDetail`

File: src/OneEntry.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    const id = useParams()
```
Comment (bowen9284) [inline src/OneEntry.jsx:8]: MUST FIX: You'll have to destructure the id here in order to get it

`const { id } = useParams()`

File: src/readAll.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	}
+
+	if (loading) {
+		return <p>loading</p>;
+		//if loading is true, then return the element and stop running the code
+	}
+
+	const output = prods.map((entry) => {
+        return(
+		<Box>
```
Comment (bowen9284) [inline src/readAll.jsx:27]: MUST FIX: Don't forget about the key prop. The console should be showing you an error.

File: src/readAll.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import {
+	Link,
+	Route,
+	Routes,
+	useNavigate,
+	BrowserRouter,
+} from "react-router-dom";
+
+function ReadAll() {
+	var { data: prods, loading, error } = useFetch("products");
```
Comment (bowen9284) [inline src/readAll.jsx:14]: Only use `let` or `const` :)

**Comment by bowen9284** [comment]
Here's an excerpt of the course that describes the nav bar issue you are having. The key is the `Outlet` component that is built into React Router. You'll want your `App` component to have your buttons / links for navigation and below those buttons, you'll have the Outlet component

[From live coding
](https://neit.instructure.com/courses/56465/pages/live-coding-adding-react-router-to-week-2?module_item_id=2530831)

```
// App.jsx
return (
  <div className="App">
    <hr />
    <button className="button" onClick={handleShowUseStateDemo}>
      UseState Demo
    </button>
    <button className="button" onClick={handleShowUseEffectDemo}>
      UseEffect Demo
    </button>
    <Outlet />
  </div>
);
```

Notice how we put the Outlet below our buttons. When we run our app again we can see that pressing the buttons renders the component linked to our path underneath of our buttons.

There's only one more thing we need to do before our new app looks like the old one. We need show our useState component by default. We'll go back to our index route which tells React Router what component we want to display by default. We will not get an error this time because the index is not a parent component.

```
ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<App />}>
          <Route index element={<UseStateDemo />} />
          <Route path="state-demo" element={<UseStateDemo />} />
          <Route path="effect-demo" element={<UseEffectDemo />} />
        </Route>
      </Routes>
    </BrowserRouter>
  </React.StrictMode>
);
```

**Review by bowen9284 — COMMENTED** [review]
I left a bunch of comments for you so hopefully it's helpful and not too overwhelming. It's hard to give you a grade because a lot of the assignment isn't quite working yet but I'm confident that you'll get full credit because you are pretty close.

**Comment by spenc-neit** [comment]
You weren't kidding, I really was close, took me only an hour to get everything working
New pull request coming soon

**Pull request merged**